src/arc.ts

```diff
@@ -653,16 +674,16 @@ export class Arc extends GraphNodeObserver {
     return descriptionHash
   }
 
-  public approveTokens(tokenAddress: Address, spender: Address, amount: BN) {
-    const signer =  (this.web3 as JsonRpcProvider).getSigner(this.defaultAccount as any)
+  public approveTokens(tokenAddress: Address, spender: Address, amount: BigNumber) {
+    const signer =  (this.web3 as providers.JsonRpcProvider).getSigner(this.defaultAccount as any)
```

Comment:
Why do you have to cast here?  IF you have to cast, would it be better that it be to `BaseProvider`?

Reply:
Firstly I try to make the obvious changes to make arc.js work and the tests pass.